### PR TITLE
check-format.pl: various small improvements, new --eol-comment option

### PR DESCRIPTION
**Pull request by DDvO**
Due to review comment https://github.com/openssl/openssl/pull/15029#discussion_r620287437 that I received recently,
added to the list in #10725 the rule
* In comparisons with constants (including `NULL`) the constant should be on the right-hand side of the comparison operator.

and added a corresponding check to `check-format.pl`.

Moreover,
* Fix false positive on struct/union/enum in func return type
* Fix false positive "no SPC before binary '*'" for '!*'
* Allow extra space before end-of-line comments unless `-e` (or equivalent `--eol-comment`) option given
* Replace 'SPC' and 'spc' by 'space' in reports and option names
* Rename `*-cmt` options to `*-comment`
* Rename "one-letter" to "single-letter", do not report variable name '`l`' (el)

- [x] tests are added or updated

### CODE REVIEW

**Comment by levitte** [comment]
What was it that got you to add this rule?
(I fear that check-format.pl is becoming incredibly picky...)

**Comment by DDvO** [comment]
> What was it that got you to add this rule?

This comment: https://github.com/openssl/openssl/pull/15029#discussion_r620287437

> (I fear that check-format.pl is becoming incredibly picky...)

There is the general problem that the list at https://github.com/openssl/openssl/issues/10725#issue-543528689 is not finalized/approved.

**Comment by DDvO** [comment]
> (I fear that check-format.pl is becoming incredibly picky...)

I think most of the possibly unwanted pickiness is regarding whitespace nits.
The tool already had various options that provide more sloppiness,
and I've just extended this PR to report needless intermediate intermediate multiple consecutive space only on `-e` (or the equivalent `--extra-spc`).

**Comment by DDvO** [comment]
@levitte, thanks for your further feedback on issues with the tool you gave in https://github.com/openssl/openssl/issues/10725#issuecomment-828739220 etc.
This PR meanwhile fixes all of them, such that `check-format.pl` no more gives false positives for the example input files you mentioned there:
```
crypto/core_namemap.c
crypto/core_fetch.c
crypto/provider_core.c 
```

File: util/check-format.pl [hunk 1]
```diff
@@ -13,8 +13,9 @@
 #
 # usage:
 #   check-format.pl [-l|--sloppy-len] [-l|--sloppy-bodylen]
-#                   [-s|--sloppy-spc] [-c|--sloppy-cmt] [-m|--sloppy-macro]
-#                   [-h|--sloppy-hang] [-1|--1-stmt]
+#                   [-s|--sloppy-spc] [-c|--sloppy-cmt]
```
Comment (paulidale) [inline util/check-format.pl:16]: It's `--sloppy-space` in the script (now).

To be honest, I'd prefer long names, so `--sloppy-comment` too.

Comment (DDvO) [reply util/check-format.pl]: > It's `--sloppy-space` in the script (now).

Oops, made consistent again.

> To be honest, I'd prefer long names, so `--sloppy-comment` too.

Yes, this is more readable. Changed as well.

**Comment by h-vetinari** [comment]
From #10725:
> > @richsalz: If the intent is that newcomers be able to use this tool, then abbreviations like "SPC" should not be used.

> @DDvO: Good point - changed to "space" (though "whitespace" would be more correct, but a bit lengthy).

This PR still leaves (and even introduces)  some `SPC` / `spc` that should IMO all be "space".

**Comment by DDvO** [comment]
> This PR still leaves (and even introduces) some `SPC` / `spc` that should IMO all be "space".

Those `SPC` occurrences are just internal, in comments and symbol names.

File: util/check-format-test-negatives.c [hunk 2]
```diff
@@ -150,6 +150,10 @@ int f(void) /*
         hanging_stmt;
 }
 
+/* not: constant on LHS of comparison or assignment operator */
```
Comment (kaduk) [inline util/check-format-test-negatives.c:153]: drive-by remark: this formatting is more often used for "note:", so if the intent is to say that this is not supposed to trigger the "constant on LHS" check, maybe write it as something like "regression test: do not trigger constant on LHS of comparison or assignment operator here"

Comment (DDvO) [reply util/check-format-test-negatives.c]: Yeah, the "not: " was a bit hard to understand here. 
Changed to "should not trigger", but did not include "regression test" here because
essentially all of `util/check-format-test-negatives.c` is some kind of regression test.

**Comment by DDvO** [comment]
Is anyone willing to approve this now?
As usual with this tool, an in-depth review is not expected :wink:

**Comment by DDvO** [comment]
Yet of course everyone is invited to further try/use the tool and give feedback :smile:

**Comment by DDvO** [comment]
Can we finalize this, PR, too, 
such that we can take advantage of these tool improvements?

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks @paulidale for your approval
